dali/c_api/c_api.h

```diff
@@ -69,6 +75,44 @@ extern "C" {
       int cpu_prefetch_queue_depth,
       int gpu_prefetch_queue_depth);
 
+  /**
+   * @brief Feed the data to ExternalSource as contiguous memory.
+   *
+   * @param pipe_handle Pointer to pipeline handle
+   * @param name Pointer to a null terminated string with the name of the External Source to be fed
+   * @param device Device of the supplied memory. Only CPU is supported.
+   * @param data_ptr Pointer to contiguous buffer containing all samples
+   * @param data_type Type of the provided data
+   * @param shape Pointer to an array containing shape of all samples concatenated one after
+   *              another. Should contain batch_size * sample_dim elements.
+   * @param sample_dim The dimensionality of the data.
+   * @param layout_str Optional layout provided as a pointer null terminated string.
+   *                   Can be set to NULL.
+   */
+  DLL_PUBLIC void daliSetExternalInput(daliPipelineHandle* pipe_handle, const char* name,
+                                       device_type_t device, const void* data_ptr,
+                                       dali_data_type_t data_type, const int64_t* shape,
+                                       int sample_dim, const char* layout_str);
+
+  /**
+   * @brief Feed the data to ExternalSource as a set of separate buffers.
+   *
+   * @param pipe_handle Pointer to pipeline handle
+   * @param name Pointer to a null terminated string with the name of the External Source to be fed
+   * @param device Device of the supplied memory. Only CPU is supported.
+   * @param data_ptr Pointer to an array containing batch_size pointers to separate Tensors.
+   * @param data_type Type of the provided data
+   * @param shape Pointer to an array containing shape of all samples concatenated one after
+   *              another. Should contain batch_size * sample_dim elements.
+   * @param sample_dim The dimensionality of the data.
+   * @param layout_str Optional layout provided as a pointer null terminated string.
```

Comment:
```suggestion
   * @param layout_str Optional layout provided as a null terminated string.
```

Reply:
What about?
```suggestion
   * @param layout_str Optional layout provided as a pointer to null terminated string.
```

Reply:
If we want to be precise here, looking at most probable usage: `daliSetExternalInputTensors(handl, "input", ...)`, it's actually a string, not really a pointer ;) But yeah, pick whichever suits you better

Reply:
I'll stick to what is done at cppreference: https://en.cppreference.com/w/cpp/string/byte/strcmp
```pointers to the null-terminated byte strings to compare```